include/picongpu/particles/ionization/byField/IonizationCurrent/JIonizationAssignment.hpp

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2020 Jakob Trojok
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/particles/ParticlesFunctors.hpp"
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/fields/FieldJ.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+    /** defining traits for current assignment
+     *
+     * \tparam T_DestSpecies type or name as boost::mpl::string of the electron species to be created
+     */
+    template< typename T_DestSpecies >
+    struct JIonizationAssignmentParent
+    {
+        using Shape = typename ::picongpu::traits::GetShape<T_DestSpecies>::type;
+        using AssignmentFunction = typename Shape::ChargeAssignmentOnSupport;
+        static constexpr int supp = AssignmentFunction::support;
+        /*(supp + 1) % 2 is 1 for even supports else 0*/
+        static constexpr int begin = -supp / 2 + (supp + 1) % 2;
+        static constexpr int end = begin+supp-1;
+    };
+
+    /**@{*/
+    /** implementation of current assignment
+     *
+     * \tparam T_Acc alpaka accelerator type
+     * \tparam T_DestSpecies type or name as boost::mpl::string of the electron species to be created
+     * \tparam T_Dim dimension of simulation
+     */
+    template<
+        typename T_Acc,
+        typename T_DestSpecies,
+        unsigned T_Dim
+    >
+    struct JIonizationAssignment;
+
+    /** 3d case
+     */
+    template<
+        typename T_Acc,
+        typename T_DestSpecies
+    >
+    struct JIonizationAssignment<T_Acc, T_DestSpecies, DIM3> : public JIonizationAssignmentParent<T_DestSpecies>
```

Comment:
I think we should try to generalize the implementation of 2d and 3d cases as possible to avoid full duplication. I could help there.

Reply:
Discussed offline. It turned out not as easy to write generically as I originally thought. So I think it's fine to keep separately as is now.